narwhals/translate.py

```diff
@@ -223,11 +225,32 @@ def get_native_namespace(obj: Any) -> Any:
     return obj.__native_namespace__()
 
 
+def narwhalify(
+    func: Callable[..., Any] | None = None, *, eager_only: bool = False
```

Comment:
todo - add the other args, not just `eager_only`. they should probably be explicitly listed for tab-completion purposes. there's not many anyway